doc/api/domain.markdown

```diff
@@ -2,304 +2,265 @@
-  // the worker
-  //
-  // This is where we put our bugs!
-
-  const domain = require('domain');
-
-  // See the cluster documentation for more details about using
-  // worker processes to serve requests.  How it works, caveats, etc.
-
-  const server = require('http').createServer((req, res) => {
-    var d = domain.create();
-    d.on('error', (er) => {
-      console.error('error', er.stack);
-
-      // Note: we're in dangerous territory!
-      // By definition, something unexpected occurred,
-      // which we probably didn't want.
-      // Anything can happen now!  Be very careful!
-
-      try {
-        // make sure we close down within 30 seconds
-        var killtimer = setTimeout(() => {
-          process.exit(1);
-        }, 30000);
-        // But don't keep the process open just for that!
-        killtimer.unref();
-
-        // stop taking new requests.
-        server.close();
-
-        // Let the master know we're dead.  This will trigger a
-        // 'disconnect' in the cluster master, and then it will fork
-        // a new worker.
-        cluster.worker.disconnect();
-
-        // try to send an error to the request that triggered the problem
-        res.statusCode = 500;
-        res.setHeader('content-type', 'text/plain');
-        res.end('Oops, there was a problem!\n');
-      } catch (er2) {
-        // oh well, not much we can do at this point.
-        console.error('Error sending 500!', er2.stack);
+The better approach is to handle thrown errors gracefully by sending an error
+response to the request that triggered the error, while letting other requests
+finish in their normal time.
+
+The `domain` module can work well with the `cluster` module. For instance,
+a master process can fork a new worker whenever a worker encounters an error.
+Use of the `domain` module allows this kind of response to occur in a more
+controlled manner.
+
+For instance, in the example `cluster` application below, worker processes
+are intentionally written to randomly throw errors. When these occur, the
+worker process shuts down immediately, leading to an ungraceful exit of the
+master process:
+
+    // An example of what not to do
+    const cluster = require('cluster');
+    
+    if (cluster.isMaster) {
+      cluster.fork();
+      cluster.fork();
+    } else {
+      function randomError() {
+        var r = Math.floor(Math.random() * 11);
+        if (r >= 7) throw new Error();
+        setTimeout(randomError, 1000);
       }
-    });
+      setTimeout(randomError, 1000);
+    }
 
-    // Because req and res were created before this domain existed,
-    // we need to explicitly add them.
-    // See the explanation of implicit vs explicit binding below.
-    d.add(req);
-    d.add(res);
+The `domain` module can be introduced within the worker process to provide
+graceful error handling as in the following modified version. However, since
+the errors are simply being ignored, resources are being leaked and held on
```

Comment:
What resources are being leaked and held on to in this case? The original example used an HTTP server, and I believe that the comment about "leaking resources like crazy" referred to, among other things, network sockets. But in this new sample code, I'm not sure there's any resource leaked.

I would suggest changing the wording to make it clear that resources would have been leaked if resources had been allocated during the workers' lifetime, or changing the sample code to allocate some resource that would leak in an event of an uncaught error handled by a domain error handler that just ignores the error.